### ISSUE TITLE
release on github fails with `github.GithubException.UnknownObjectException`

### ISSUE THREAD

**Issue by olivierdalang** [opengisch/qgis-plugin-ci#135]
Hi !

Trying to the a release to github, using almost exactly the github setup from [the doc](https://opengisch.github.io/qgis-plugin-ci/usage/ci_github.html) I get the error below.

The only notable difference is that I don't provide the `--osgeo-*` arguments since I don't need to upload to the official repo. Also this is in a private repository.

I tried with simple release names (`0.0.1`), and with and without the experimental flag, but nothing helped :-(

Any idea what could be happening ? (error logs and workflow definition below)

Cheers !

Olivier

Error logs
```
Run qgis-plugin-ci release --github-token *** --create-plugin-repo ${GITHUB_REF/refs\/tags\//}
  qgis-plugin-ci release --github-token *** --create-plugin-repo ${GITHUB_REF/refs\/tags\//}
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.13/x64
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.13/x64/bin/qgis-plugin-ci", line 11, in <module>
    load_entry_point('qgis-plugin-ci==[2](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:2).[3](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:3).0', 'console_scripts', 'qgis-plugin-ci')()
  File "/opt/hostedtoolcache/Python/3.7.13/x6[4](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:4)/lib/python3.7/site-packages/scripts/qgis_plugin_ci.py", line 203, in main
    disable_submodule_update=args.disable_submodule_update,
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/qgispluginci/release.py", line 477, in release
    parameters, release_tag=release_tag, github_token=github_token
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/qgispluginci/release.py", line 304, in release_is_prerelease
    repo = Github(github_token).get_repo(slug)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/github/MainClass.py", line 330, in get_repo
    headers, data = self.__requester.requestJsonAndCheck("GET", url)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/github/Requester.py", line 3[5](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:5)5, in requestJsonAndCheck
    verb, url, parameters, headers, input, self.__customConnection(url)
  File "/opt/hostedtoolcache/Python/3.7.13/x[6](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:7)4/lib/python3.[7](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:8)/site-packages/github/Requester.py", line 37[8](https://github.com/opengisch/ti-autoigd-lib/runs/7115800655?check_suite_focus=true#step:5:9), in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-repository"}
Error: Process completed with exit code 1.
```

Workflow definition
```yml
name: Release

on:
  release:
    types: published

jobs:
  release:
    name: Release
    runs-on: ubuntu-18.04

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup python
        uses: actions/setup-python@v1
        with:
          python-version: 3.7

      - name: Install qgis-plugin-ci
        run: pip install qgis-plugin-ci~=2.3.0

      - name: Release
        run: >-
          qgis-plugin-ci release
          --github-token ${{ secrets.GITHUB_TOKEN }}
          --create-plugin-repo
          ${GITHUB_REF/refs\/tags\//}

```

**Comment by olivierdalang** [comment]
I'm now using this as a workaround instead of the release step (maybe a good idea anyway to factor uploading the assets out to an already existing action, always better to have less code to maintain)

```
      - name: Package
        run: qgis-plugin-ci package ${GITHUB_REF##*/}

      - name: Upload assets
        uses: softprops/action-gh-release@v1
        with:
          files: tiautoigdplugin.*.zip
```

**Comment by Guts** [comment]
Hi @olivierdalang,

Sorry for the delayed answer, holidays ruled!

Thanks for your report!
If I understood well, your problem is solved using another GH Action to create the GH Release instead of the official one (marked as archived)?

**Comment by olivierdalang** [comment]
Hi !

It's not solved (well... haven't tried again since) but worked around.

Instead of using `qgis-plugin-ci release ...` which I think makes a zip then uploads it, I just use `qgis-plugin-ci package ...` which only makes the zip, then upload in a separate step with the action `softprops/action-gh-release@v1`.

If it's difficult to fix/maintain, I was suggesting to completely drop the release to github functionality, and instead just document using already existing actions as I did.

Cheers !

**Comment by Guts** [comment]
`qgis-plugin-ci release` **does not create** a GH Relase **but update** an existing one. Maybe the error message was just saying there is no release matching the tag name?

**Comment by 3nids** [comment]
This tool was written at the old Travis-CI times.
I agree that the upload step could be left apart for the sake of simplicity.

**Comment by 3nids** [comment]
fixed thanks to #259
